sys/riotboot/serial.c

```diff
@@ -0,0 +1,267 @@
+/*
+ * Copyright (C) 2020 Benjamin Valentin
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @ingroup     sys_riotboot_serial
+ * @{
+ *
+ * @file
+ * @brief       Serial Bootloader
+ *
+ * @author      Benjamin Valentin <benjamin.valentin@ml-pa.com>
+ *
+ * @}
+ */
+
+#include <assert.h>
+#include <string.h>
+
+#include "stdio_uart.h"
+#include "periph/uart.h"
+#include "periph/flashpage.h"
+#include "unaligned.h"
+#include "checksum/crc8.h"
+#include "riotboot/serial.h"
+#include "riotboot/magic.h"
+
+#include "board.h"
+
+#ifndef RIOTBOOT_UART_DEV
+#define RIOTBOOT_UART_DEV       STDIO_UART_DEV
+#endif
+
+#ifndef RIOTBOOT_UART_BAUDRATE
+#define RIOTBOOT_UART_BAUDRATE  STDIO_UART_BAUDRATE
```

Comment:
We might want to default to something faster here, e.g. 576000 baud

Reply:
Will all platforms support it?

Reply:
well let's keep it like this for now, we can always configure it on a per-board basis.